matrix-api.c

```diff
@@ -420,9 +422,10 @@ static gchar *_build_request(PurpleAccount *acct, const gchar *url,
     g_string_append_printf(request_str, "Host: %.*s\r\n",
             (int)(url_path-url_host), url_host);
 
+    g_string_append(request_str, extra_headers);
```

Comment:
does this behave sensibly if extra_headers is NULL? Should probably make it do so if not.